sql_lineage/lineage.py

```diff
@@ -16,8 +18,12 @@ def _parse_ctes(sql: str) -> [Nodes, Edges]:
     """
     Parse an SQL query into nodes and edges based on CTEs.
     """
-    parsed = qualify.qualify(sqlglot.parse_one(sql))
     nodes, edges = [], []
+    parsed = sqlglot.parse_one(sql)
+    with contextlib.suppress(Exception):
+        # There isn't a hard requirement for the query to be valid SQL,
+        # but it's better if it is
+        parsed = qualify.qualify(parsed)
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Consider specifying exception types in contextlib.suppress.

Catching a generic Exception could suppress unexpected errors, making debugging harder. It's better to catch specific exceptions that you anticipate.

```suggestion
        select *, :something
        from ccc
            inner join ddd using (aa)
```

Reply:
Is this comment correct?

Reply:
Is this comment helpful?

Reply:
Is the comment type correct?

Reply:
Is the comment area correct?